bids/layout/layout.py

```diff
@@ -290,12 +298,12 @@ def __getattr__(self, key):
                 if sing in entities:
                     ent_name = sing
                 else:
-                    raise AttributeError(
+                    raise BIDSEntityError(
                         "'get_{}' can't be called because '{}' isn't a "
                         "recognized entity name.".format(ent_name, ent_name))
             return partial(self.get, return_type='id', target=ent_name)
         # Spit out default message if we get this far
-        raise AttributeError("%s object has no attribute named %r" %
+        raise BIDSEntityError("%s object has no attribute named %r" %
                              (self.__class__.__name__, key))
```

Comment:
I don't think this is expecting an entity but is a more typical `AttributeError`.

```suggestion
        raise AttributeError("%s object has no attribute named %r" %
                             (self.__class__.__name__, key))
```